schema/bom-1.3-SNAPSHOT.proto

```diff
@@ -0,0 +1,391 @@
+  // Community or commercial support
+  EXTERNAL_REFERENCE_TYPE_SUPPORT = 10;
+  // Direct or repository download location
+  EXTERNAL_REFERENCE_TYPE_DISTRIBUTION = 11;
+  // The URL to the license file. If a license URL has been defined in the license node, it should also be defined as an external reference for completeness
+  EXTERNAL_REFERENCE_TYPE_LICENSE = 12;
+  // Build-system specific meta file (i.e. pom.xml, package.json, .nuspec, etc)
+  EXTERNAL_REFERENCE_TYPE_BUILD_META = 13;
+  // URL to an automated build system
+  EXTERNAL_REFERENCE_TYPE_BUILD_SYSTEM = 14;
+}
+
+enum HashAlg {
+  HASH_ALG_NULL = 0;
+  HASH_ALG_MD_5 = 1;
+  HASH_ALG_SHA_1 = 2;
+  HASH_ALG_SHA_256 = 3;
+  HASH_ALG_SHA_384 = 4;
+  HASH_ALG_SHA_512 = 5;
+  HASH_ALG_SHA_3_256 = 6;
+  HASH_ALG_SHA_3_384 = 7;
+  HASH_ALG_SHA_3_512 = 8;
+  HASH_ALG_BLAKE_2_B_256 = 9;
+  HASH_ALG_BLAKE_2_B_384 = 10;
+  HASH_ALG_BLAKE_2_B_512 = 11;
+  HASH_ALG_BLAKE_3 = 12;
+}
+
+// Specifies the file hash of the component
+message Hash {
+  // Specifies the algorithm used to create the hash
+  HashAlg alg = 1;
+  // SimpleContent value of element
+  string value = 2;
+}
+
+message IdentifiableAction {
+  // The timestamp in which the action occurred
+  uint64 timestamp = 1;
+  // The name of the individual who performed the action
+  string name = 2;
+  // The email address of the individual who performed the action
+  string email = 3;
+}
+
+enum IssueClassification {
+  ISSUE_CLASSIFICATION_NULL = 0;
+  // A fault, flaw, or bug in software
+  ISSUE_CLASSIFICATION_DEFECT = 1;
+  // A new feature or behavior in software
+  ISSUE_CLASSIFICATION_ENHANCEMENT = 2;
+  // A special type of defect which impacts security
+  ISSUE_CLASSIFICATION_SECURITY = 3;
+}
+
+message Issue {
+  // Specifies the type of issue
+  IssueClassification type = 1;
+  // The identifier of the issue assigned by the source of the issue
+  string id = 2;
+  // The name of the issue
+  string name = 3;
+  // A description of the issue
+  string description = 4;
+  Source source = 5;
+  repeated string references = 6;
+}
+
+// The source of the issue where it is documented.
+message Source {
+  // The name of the source. For example "National Vulnerability Database", "NVD", and "Apache"
+  string name = 1;
+  // The url of the issue documentation as provided by the source
+  string url = 2;
+}
+
+message ComponentLicense {
+  License license = 1;
+  string expression = 2;
+}
+
+message License {
+  // A valid SPDX license ID
+  string id = 1;
+  // If SPDX does not define the license used, this field may be used to provide the license name
+  string name = 2;
+  // Specifies the optional full text of the attachment
+  AttachedText text = 3;
+  // The URL to the attachment file. If the attachment is a license or BOM, an externalReference should also be specified for completeness.
+  string url = 4;
+}
+
+message Metadata {
+  // The date and time (timestamp) when the document was created.
+  uint64 timestamp = 1;
```

Comment:
You can alternatively use `google.protobuf.Timestamp` here, and import it `import "google/protobuf/timestamp.proto";`, 

it's been available since proto3 and might add more flexibility/stronger typing (the `uint64` doesn't force a UNIX timestamp, but encourages it)

Reply:
Thanks @DarthHater. Is that pretty much universally supported across protobuf implementations?

Reply:
For `proto3` I believe so! Being you specified `proto3` you should be good to go! This stackoverflow post is where I'm gleaning some of that info from, if you want to chase it further: https://stackoverflow.com/a/42922721/338597

Reply:
Thanks @DarthHater, I’ll do some further work on the proto file in a seperate PR and tag you in it.

Reply:
New version uses `google.protobuf.Timestamp`